plugins/block-plus-minus/test/if.mocha.js

```diff
@@ -27,134 +37,201 @@ suite('if & ifelse', () => {
+        createBlock: trivialCreateBlock},
+    ];
+    const jsSuiteTestCases = [
+      {title: 'Trivial', expectedCode: 'if (false) {\n}\n',
+        createBlock: trivialCreateBlock},
+    ];
+    const luaSuiteTestCases = [
+      {title: 'Trivial', expectedCode: 'if false then\nend\n',
+        createBlock: trivialCreateBlock},
+    ];
+    const phpSuiteTestCases = [
+      {title: 'Trivial', expectedCode: 'if (false) {\n}\n',
+        createBlock: trivialCreateBlock},
+    ];
+    const pythonSuiteTestCases = [
+      {title: 'Trivial', expectedCode: 'if False:\nundefined',
+        createBlock: trivialCreateBlock},
+    ];
+    const testSuites = [
+      {title: 'Dart', generator: Blockly.Dart, testCases: dartSuiteTestCases},
+      {title: 'JavaScript', generator: Blockly.JavaScript,
+        testCases: jsSuiteTestCases},
+      {title: 'Lua', generator: Blockly.Lua, testCases: luaSuiteTestCases},
+      {title: 'PHP', generator: Blockly.PHP, testCases: phpSuiteTestCases},
+      {title: 'Python', generator: Blockly.Python,
+        testCases: pythonSuiteTestCases},
+    ];
+
+    const createGeneratorTestFn = (generator) => {
+      return (testCase) => {
+        return function() {
+          const block = testCase.createBlock();
+          const code = generator.blockToCode(block);
+          assert.equal(code, testCase.expectedCode);
+        };
+      };
+    };
+
+    testSuites.forEach(function(suiteInfo) {
+      suite(suiteInfo.title, function() {
+        runTestCases(
+            suiteInfo.testCases, createGeneratorTestFn(suiteInfo.generator));
+      });
     });
-    test('No else', () => {
-      const oldText = '<xml xmlns="https://developers.google.com/blockly/xml">\n' +
-          '  <block type="controls_if" id="if" x="44" y="134">\n' +
-          '    <mutation elseif="2"></mutation>\n' +
-          '  </block>\n' +
-          '</xml>';
-      Blockly.Xml.domToWorkspace(
-          Blockly.Xml.textToDom(oldText), this.workspace);
-      assertIf(this.workspace.getBlockById('if'), 3);
-      const xml = Blockly.Xml.workspaceToDom(this.workspace);
-      const newText = Blockly.Xml.domToPrettyText(xml);
-      assert.equal(newText, oldText);
+  });
+
+  suite('Serialization', function() {
+    function assertXmlMatch(xml, blockType, expectedBlockContents='') {
+      const expectedXml =
+          new RegExp('<block[^>]* type="' + blockType +
+              '" [^>]*>\\s*' + expectedBlockContents + '\\s*</block>');
+      assert.match(xml, expectedXml);
+    }
+
+    suite('blockToXml', function() {
+
+      setup(function() {
+        this.block = this.workspace.newBlock('controls_if');
+      });
+
+      test('Trivial', function() {
+        const xml =
+            Blockly.Xml.domToPrettyText(Blockly.Xml.blockToDom(this.block));
+        assertXmlMatch(xml, 'controls_if');
+      });
+
+      test('No else', function() {
+        this.block.plus();
+        this.block.plus();
```

Comment:
I'm not sure if this is better or if I should instead do round trip with `domToBlock(textToDom(` like in 'With else' test